### PR TITLE
Accept new value when hitting ENTER to close a prompt dialog

### PR DESCRIPTION
**Pull request by Stormalong**
…essed

## Proposed change

This fixes a bug that causes hitting ENTER to close a prompt dialog to not accept the new value.

Steps to reproduce:
 - go to Configuration
 - go to Devices & Services
 - click the kebab menu on any card and select Rename
 - type in a new name and hit ENTER (do not click OK)
 - the dialog will close but the name will not be changed

The existing code clearly intends for hitting ENTER (keycode 13) to accept the new value (by calling _confirm) but it doesn't work because the @change event isn't called for keystrokes.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by zsarnett — DISMISSED** [review]
This will now call the value changed event each time a key is pressed..

File: src/dialogs/generic/dialog-box.ts [hunk 1]
```diff
@@ -76,7 +76,7 @@ class DialogBox extends LitElement {
                 <ha-textfield
                   dialogInitialFocus
                   .value=${this._value || ""}
-                  @keyup=${this._handleKeyUp}
+                  @keyup=${this._valueChanged}
```
Comment (zsarnett) [inline src/dialogs/generic/dialog-box.ts:79]:
```suggestion
                  @keyup=${this._handleKeyUp}
```

File: src/dialogs/generic/dialog-box.ts [hunk 2]
```diff
@@ -120,12 +120,6 @@ class DialogBox extends LitElement {
     this._close();
   }
 
-  private _handleKeyUp(ev: KeyboardEvent) {
-    if (ev.keyCode === 13) {
-      this._confirm();
-    }
-  }
-
   private _confirm(): void {
```
Comment (zsarnett) [inline src/dialogs/generic/dialog-box.ts:123]:
```suggestion
 private _handleKeyUp(ev: KeyboardEvent) {
    if (ev.keyCode === 13) {
      this._confirm();
    }
  }
  
  private _confirm(): void {
```

File: src/dialogs/generic/dialog-box.ts [hunk 3]
```diff
@@ -76,7 +76,7 @@ class DialogBox extends LitElement {
                 <ha-textfield
                   dialogInitialFocus
                   .value=${this._value || ""}
-                  @keyup=${this._handleKeyUp}
+                  @keyup=${this._valueChanged}
                   @change=${this._valueChanged}
```
Comment (zsarnett) [inline src/dialogs/generic/dialog-box.ts:80]:
```suggestion
                  @input=${this._valueChanged}
```

Comment (zsarnett) [reply src/dialogs/generic/dialog-box.ts:80]: This will update the changed value ,immediately

Comment (Stormalong) [reply src/dialogs/generic/dialog-box.ts]: Suggestions applied

**Review by zsarnett — APPROVED** [review]

**Review by steverep — COMMENTED** [review]
I noticed this issue as well.  Someone else reported in #12206.  Per my comments, a better fix should be just to get rid of the key up and value changed handler methods, and grab the current field value in the confirm method, saving a bunch of lines of code.

**Pull request merged**